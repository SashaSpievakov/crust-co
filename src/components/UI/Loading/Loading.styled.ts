import styled, { keyframes } from 'styled-components';

const rotate = keyframes`
  100% {
    transform: rotate(360deg);
  }
`;

const dash = keyframes`
  0% {
    stroke-dasharray: 1, 150;
    stroke-dashoffset: 0;
  }
  50% {
    stroke-dasharray: 90, 150;
    stroke-dashoffset: -35;
  }
  100% {
    stroke-dasharray: 90, 150;
    stroke-dashoffset: -124;
  }
`;

const Wrapper = styled.section`
  width: 100px;
  margin: 0 auto;
`;

const Loader = styled.svg`
  width: 100px;
  height: 100px;
  margin: 250px auto;
  animation: ${rotate} 2s linear infinite;
`;

const Circle = styled.circle`
  stroke: ${({ theme }) => theme.colors.primary};
  stroke-linecap: round;
  animation: ${dash} 1.5s ease-in-out infinite;
`;

export { Wrapper, Loader, Circle };
