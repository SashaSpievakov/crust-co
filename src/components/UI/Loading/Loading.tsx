import { Wrapper, Loader, Circle } from './Loading.styled';

const Loading = () => {
  return (
    <Wrapper>
      <Loader viewBox="0 0 50 50">
        <Circle cx="25" cy="25" r="20" fill="none" strokeWidth="4" />
      </Loader>
    </Wrapper>
  );
};

export default Loading;
