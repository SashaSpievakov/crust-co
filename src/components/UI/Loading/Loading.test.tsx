import '@testing-library/jest-dom';

import rendererWithAllProviders from '../../../tests/helpers/rendererWithProviders';
import Loading from './Loading';

describe('Loading Tests', () => {
  test('renders the Loading UI component', () => {
    const snapshot = rendererWithAllProviders(<Loading />);
    expect(snapshot).toMatchSnapshot();
  });
});
