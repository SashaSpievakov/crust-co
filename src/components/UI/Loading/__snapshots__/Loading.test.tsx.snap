// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading tests should render the Loading UI component 1`] = `
Object {
  "children": Array [
    <svg
      className="sc-ksBlkl DtAzl"
      viewBox="0 0 50 50"
    >
      <circle
        className="sc-hBxehG eqkYYb"
        cx="25"
        cy="25"
        fill="none"
        r="20"
        strokeWidth="4"
      />
    </svg>,
  ],
  "props": Object {
    "className": "sc-bqWxrE dHOchB",
  },
  "type": "article",
}
`;
